src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -153,14 +153,20 @@ public static ITypeSymbol RemoveNullableIfPresent(this ITypeSymbol symbol)
             // instantiations of that method.
             var originalInterfaceType = interfaceMember.ContainingType.OriginalDefinition;
             var originalInterfaceMember = interfaceMember.OriginalDefinition;
+
             var constructedInterfaces = typeSymbol.AllInterfaces.Where(i =>
                 SymbolEquivalenceComparer.Instance.Equals(i.OriginalDefinition, originalInterfaceType));
 
+            // Try to get the compilation for the symbol we're searching for, 
+            // which can help identify matches with the call to SymbolFinder.OriginalSymbolsMatch
+            var typeSymbolProject = solution.GetProject(typeSymbolAndProjectId.ProjectId);
+            var typeSymbolCompilation = typeSymbolProject?.GetCompilationAsync(cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
1. What happens if we get null for `typeSymbolCompilation` here?  
2. i would prefer we actually just move to async, and thread that backwards.

Reply:
Null `typeSymbolCompilation` is handled correctly in the following call. There are a few tests that exercise this by not providing a projectId. 

I'll look at how much work it is to push Async up.

Reply:
Async streams aren't available yet, so this is fine as is for now. I don't think it's worth changing to be less lazy for this one call.

Reply:
i'm not asking for an async-stream, just an async method.  i.e.  return a `Task<ImmutableArray<Whatever>>` here.

Reply:
Good point. There's no real reason for this to yield in a lazy manner. I'll change to `Task<ImmutableArray<>>`